sample/kotlin/src/main/kotlin/com/datadog/android/sample/datalist/DataListViewModel.kt

```diff
@@ -8,36 +8,56 @@ package com.datadog.android.sample.datalist
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import com.datadog.android.rx.publishErrorsToRum
 import com.datadog.android.sample.data.DataRepository
-import com.datadog.android.sample.data.Result
 import com.datadog.android.sample.data.model.Log
-
-class DataListViewModel(val repository: DataRepository) : ViewModel(),
-    DataRepository.Callback<List<Log>> {
-
-    val data = mutableListOf<String>()
-    val liveData = MutableLiveData<UIResponse>()
-
-    fun onAddData() {
-        data.add("Item ${data.size}")
-        repository.getLogs("source:android", this)
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.subjects.PublishSubject
+
+class DataListViewModel(val repository: DataRepository) : ViewModel() {
+
+    private val uiRequestSubject: PublishSubject<UIRequest> = PublishSubject.create()
+    private var disposable: Disposable? = null
```

Comment:
Best practices for viewModels are: you should listen to the disposable in the on resume, push the new value in your `PublishSubject` in your `performRequest()` method, and `dispose` of your observable when the fragment pauses. That way your fragment will always display the latest info, will keep listening and you can then perform multiple successive requests if you want.

Reply:
Yes I know but for this you need a `BehaviorSubject` or `ReplaySubject`, instead what I am doing there I am using RxJava together with LiveData ....basically is the LiveData responsible to deliver the `UIResponse` to the Fragment and this will depend on the `viewLifecycleObserver`. Now I actually assume that `LiveData` keeps and delivers actually the last value to the View on subscription :)